gensim/models/_utils_any2vec.pyx

```diff
@@ -21,26 +25,28 @@ cdef _byte_to_int_py2(b):
 _byte_to_int = _byte_to_int_py2 if PY2 else _byte_to_int_py3
 
 
-cpdef ft_hash(unicode string):
-    """Calculate hash based on `string`.
+cpdef ft_hash_bytes(bytes bytez):
+    """Calculate hash based on `bytez`.
     Reproduce `hash method from Facebook fastText implementation
     <https://github.com/facebookresearch/fastText/blob/master/src/dictionary.cc>`_.
 
     Parameters
     ----------
-    string : unicode
-        The string whose hash needs to be calculated.
+    bytez : bytes
+        The string whose hash needs to be calculated, encoded as UTF-8.
 
     Returns
     -------
     unsigned int
         The hash of the string.
 
     """
-    cdef unsigned int h = 2166136261
-    for c in string.encode("utf-8"):
-        h = np.uint32(h ^ np.uint32(np.int8(_byte_to_int(c))))
-        h = np.uint32(h * np.uint32(16777619))
+    cdef uint32_t h = 2166136261
+    cdef char b
+
+    for b in bytez:
+        h = h ^ <uint32_t>(<int8_t>b)  # FIXME I drop 'ord' from py2, not sure about correctenss
```

Comment:
what's about `ord`?

Reply:
ping @mpenkov

Reply:
it doesn't affect the results of the unit tests, so I think we can forget about it.